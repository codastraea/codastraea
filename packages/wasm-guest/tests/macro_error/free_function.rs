use serpent_automation_wasm_guest::workflow;

pub struct X;

impl X {
    #[workflow]
    async fn counter(&self) {}
}

fn main() {}
