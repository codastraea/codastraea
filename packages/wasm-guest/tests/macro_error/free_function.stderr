error: `workflow` functions should have no runtime parameters
 --> tests/macro_error/free_function.rs:7:22
  |
7 |     async fn counter(&self) {}
  |                      ^^^^^

error: `workflow` functions should be free functions, not methods
 --> tests/macro_error/free_function.rs:7:22
  |
7 |     async fn counter(&self) {}
  |                      ^^^^^
